src/components/Dashboard/DirectorListSm.vue

```diff
@@ -87,10 +82,23 @@ export default class DirectorListSm extends Mixins(CommonMixin, CountriesProvinc
   @Prop({ default: false })
   readonly showGrayedOut: boolean
 
-  @State directors!: Array<object>
+  @Getter getParties!: PartyIF[]
+
+  /** The directors list. */
+  get directors (): DirectorIF[] {
+    const directors = this.getParties.filter(
+      party => party.roles?.some(role => role.roleType === Roles.DIRECTOR)
+    )
+    const directorsWithProps = directors.map((director: any, i: number) =>
+      ({ ...director, id: i + 1 } as DirectorIF)
+    )
+    return directorsWithProps.sort(
+      this.fieldSorter(['lastName', 'firstName', 'middleName'])
+    )
+  }
```

Comment:
I didn't make this a global getters since this is the only place where we need to extract directors from the parties list.